common/persistence/client/quotas.go

```diff
@@ -30,23 +30,57 @@ import (
 )
 
 var (
-	CallerTypePriority = map[string]int{
-		headers.CallerTypeAPI:        0,
-		headers.CallerTypeBackground: 1,
+	CallerTypeDefaultPriority = map[string]int{
+		headers.CallerTypeAPI:        1,
+		headers.CallerTypeBackground: 2,
 	}
 
-	APIPriorityOverride = map[string]int{
+	APITypeCallInitiationPriorityOverride = map[string]int{
+		"StartWorkflowExecution":           0,
+		"SignalWithStartWorkflowExecution": 0,
+		"SignalWorkflowExecution":          0,
+		"RequestCancelWorkflowExecution":   0,
+		"TerminateWorkflowExecution":       0,
+		"GetWorkflowExecutionHistory":      0,
+		"UpdateWorkflow":                   0,
+	}
+
+	BackgroundTypeAPIPriorityOverride = map[string]int{
 		"GetOrCreateShard": 0,
 		"UpdateShard":      0,
 
 		// this is a preprequisite for checkpoint queue process progress
 		"RangeCompleteHistoryTasks": 0,
 	}
 
-	RequestPrioritiesOrdered = []int{0, 1}
+	RequestPrioritiesOrdered = []int{0, 1, 2}
 )
 
 func NewPriorityRateLimiter(
+	namespaceMaxQPS PersistenceNamespaceMaxQps,
+	hostMaxQPS PersistenceMaxQps,
+) quotas.RequestRateLimiter {
+	hostRequestRateLimiter := newPriorityRateLimiter(
+		func() float64 { return float64(hostMaxQPS()) },
+	)
+
+	return quotas.NewNamespaceRateLimiter(func(req quotas.Request) quotas.RequestRateLimiter {
+		if req.Caller != "" && req.Caller != headers.CallerNameSystem {
```

Comment:
this means system will have higher priority compare to user namespace

Reply:
I am not following. System calls go through one ratelimiter. User namespace calls first go through the a per ns ratelimiter, and then the same ratelimiter as the system calls. So as long as user calls are within their limit, I don't see why they will get lower priority.